### ISSUE TITLE
Move image desription to bottom

### ISSUE THREAD

**Issue by philipzae** [IBBoard/cawbird#293]
How it currently looks and you have to mouse over it to see the description
![image](https://user-images.githubusercontent.com/4412114/105198191-d7c10b80-5b56-11eb-9e0b-739ae76716ea.png)

How it could look and would always be visible
![image](https://user-images.githubusercontent.com/4412114/105199743-8ade3480-5b58-11eb-9a7d-2976b591d820.png)

**Comment by IBBoard** [comment]
It's a tooltip, so we don't have much control over where it goes. Gnome puts it wherever you leave your mouse.

If we make it a label then a) we've got to control the time for it fading in and out, which tooltips do automatically and b) I'm not sure it integrates as well with accessibility tools.

That said, maybe @CodedOre has some ideas to go alongside #193?

**Comment by philipzae** [comment]
> Edit: After posting that, I realised that I only half-read the content (I blame too much text in the images!) and you mean "show it all the time" rather than "have it as a tool tip". I guess that works, but it is mainly there as an accessibility thing. Most people don't use it. A lot of people who use it don't include anything worthwhile. And a lot of GIFs just have something like "[search-word] GIF" as their description.

Yes that is what I meant. For the times I've seen the text used, they are using it as alt text (all the text found in the images is in it) and GIFs having that text would be the same. Ideally accessibility tools should read 'Images 1 of 2' and the description.

**Comment by CodedOre** [comment]
> That said, maybe @CodedOre has some ideas to go alongside #193?

I only took a small look, but it seems like Twitter only has image description on images, not on videos.

So I think it should work to just replace video playback with image description, when I redesign the media player with the new design from my mockup in #177:

![img-desc](https://user-images.githubusercontent.com/47981497/105343402-e0125880-5be1-11eb-81f6-93e869a62c1c.png)

Otherwise we can just place the description over the playback.

**Comment by IBBoard** [comment]
Videos don't have alt text in quite the same way, but they do support [titles and descriptions](https://github.com/IBBoard/cawbird/blob/master/src/CbTypes.c#L510-L515), which we load as alt text and display. They seem to be used most often on news articles and other more Corporate media.

I've just tried adding a description to a video and it looks like Twitter doesn't reject it, but it doesn't actually attach it either. Which is annoying.

(Also, I hadn't paid enough attention to realise that the picture was a cup and laptop on a weird table - I'd half-seen some weird frog-shark creature!)

**Comment by philipzae** [comment]
> (Also, I hadn't paid enough attention to realise that the picture was a cup and laptop on a weird table - I'd half-seen some weird frog-shark creature!)

White table and white laptop is definitely gonna confuse people. :smile:

**Comment by IBBoard** [comment]
It's not even table shaped! Who has two coloured shapes and a white shape as a table top? 😄

Just had a further thought - is it better to make this a permanent thing (showing the text like a caption, but taking up space) or a temporary thing (using a revealer to show/hide on mouse over)?

And, since I'm being slow on the uptake with this design, I've just realised that the "put it with the image count" mock-up *should* work without much redesign because we're already putting the image count in a revealer. #facepalm

**Comment by CodedOre** [comment]
> Just had a further thought - is it better to make this a permanent thing (showing the text like a caption, but taking up space) or a temporary thing (using a revealer to show/hide on mouse over)?

My idea was to allow to hide/show the UI with a click/tap on non-UI. A mouse over might not be as intuitive, as this can happen "randomly", while when it gets hide with a mouse click you might suspect it can be undo with another click.
This behavior is also commonly used on mobile interfaces.

**Comment by IBBoard** [comment]
Maybe. Our current behaviour is to close on any click, but I guess this is turning into more of a full window with title bar.

Thinking of which, does that work on non-Gnome platforms? I'm not sure which bits "just work" and which bits become window content, and what effect that will have on content and layout.

**Comment by CodedOre** [comment]
> Maybe. Our current behaviour is to close on any click, but I guess this is turning into more of a full window with title bar.

Well, my thought was more about using `HdyWindow`, so we don't have a fixed titlebar at anytime, but only when we want.

> Thinking of which, does that work on non-Gnome platforms? I'm not sure which bits "just work" and which bits become window content, and what effect that will have on content and layout.

Well, as far as I can tell, it should work without problems on non-Gnome Platforms with `HdyWindow`, as we only modify the content of the window.

By the way, does the contact form on your site work? I would like to discuss something with you in more detail.

**Comment by IBBoard** [comment]
> Well, my thought was more about using `HdyWindow`, so we don't have a fixed titlebar at anytime, but only when we want.
> …
> Well, as far as I can tell, it should work without problems on non-Gnome Platforms with HdyWindow, as we only modify the content of the window.

I did think that it looked like "Gnome header bar as overlapping window content". It might look a bit out of place on other desktops, but then various bits of the UI probably do anyway!

> By the way, does the contact form on your site work? I would like to discuss something with you in more detail.

Yeah, I got the email (four times for some reason!). I'll reply over the weekend.

**Comment by CodedOre** [comment]
> > By the way, does the contact form on your site work? I would like to discuss something with you in more detail.
> 
> 
> 
> Yeah, I got the email (four times for some reason!). I'll reply over the weekend.

Good! Was not sure as the form did not confirm that it send the message, so excuse the spamming.
